test/room_test.exs

```diff
@@ -23,7 +23,7 @@ defmodule RoomTest do
 
     _user2 =
       spawn_user_no_join(fn ->
-        GenServer.call(room_pid, {:join, self(), %{user: "1"}})
+        GenServer.call(room_pid, {:join, self(), %{user: "1"}, 0})
```

Comment:
Mmh after reading through the code I don't understand this new parameter. In the test it is always 0. Does it vary with and without turn?

Reply:
well we have a per user turn timeout which is part of the state of the user session, so usually it's 0 when the session starts but then it will have the value of the last timout generated

Reply:
is there a way to make that clearer?

Reply:
Maybe GenServer's handle_call can have 0 as default value

Reply:
Since this is really only for the test unclear and not for the actual code, I would leave it. The parameter is clearly named now so that if you look into the handler, it should be clear what it means.